driver-kotlin-extensions/src/test/kotlin/com/mongodb/kotlin/client/model/KPropertiesTest.kt

```diff
@@ -0,0 +1,140 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mongodb.kotlin.client.model
+
+import java.util.Locale
+import kotlin.test.assertEquals
+import kotlinx.serialization.SerialName
+import org.bson.codecs.pojo.annotations.BsonId
+import org.bson.codecs.pojo.annotations.BsonProperty
+import org.junit.Test
+import org.junit.jupiter.api.assertThrows
+
+class KPropertiesTest {
+
+    data class Restaurant(
+        @BsonId val a: String,
+        @BsonProperty("b") val bsonProperty: String,
+        @SerialName("c") val serialName: String,
+        val name: String,
+        val stringList: List<String>,
+        val localeMap: Map<Locale, Review>,
+        val reviews: List<Review>,
+        @BsonProperty("nested") val subDocument: Restaurant?
+    )
+
+    data class Review(
+        @BsonProperty("prop") val bsonProperty: String,
+        @SerialName("rating") val score: String,
+        val name: String,
+        @BsonProperty("old") val previous: Review?
+    )
+
+    @Test
+    fun testPath() {
+        assertEquals("_id", Restaurant::a.path())
+        assertEquals("b", Restaurant::bsonProperty.path())
+        assertEquals("c", Restaurant::serialName.path())
+        assertEquals("name", Restaurant::name.path())
+        assertEquals("stringList", Restaurant::stringList.path())
+        assertEquals("localeMap", Restaurant::localeMap.path())
+        assertEquals("nested", Restaurant::subDocument.path())
+        assertEquals("reviews", Restaurant::reviews.path())
+
+        assertEquals("prop", Review::bsonProperty.path())
+        assertEquals("rating", Review::score.path())
+        assertEquals("name", Review::name.path())
+        assertEquals("old", Review::previous.path())
+    }
+
+    @Test
+    fun testDivOperator() {
+        assertEquals("nested._id", (Restaurant::subDocument / Restaurant::a).path())
+        assertEquals("nested.b", (Restaurant::subDocument / Restaurant::bsonProperty).path())
+        assertEquals("nested.c", (Restaurant::subDocument / Restaurant::serialName).path())
+        assertEquals("nested.name", (Restaurant::subDocument / Restaurant::name).path())
+        assertEquals("nested.stringList", (Restaurant::subDocument / Restaurant::stringList).path())
+        assertEquals("nested.localeMap", (Restaurant::subDocument / Restaurant::localeMap).path())
+        assertEquals("nested.nested", (Restaurant::subDocument / Restaurant::subDocument).path())
+    }
+
+    @Test
+    fun testRemOperator() {
```

Comment:
Curious to know the use case of these paths?

Reply:
Honestly no idea - they are mentioned here:  https://litote.org/kmongo/typed-queries/#unsafe-nested-properties-and-operator Not sure of the origins. 


I'd be happy to remove it and we can always add it back in customers desire it.

Reply:
let's keep it for the sake of compatibility and migration from KMongo